pandas/tests/series/test_cumulative.py

```diff
@@ -227,3 +229,55 @@ def test_cumprod_timedelta(self):
         ser = pd.Series([pd.Timedelta(days=1), pd.Timedelta(days=3)])
         with pytest.raises(TypeError, match="cumprod not supported for Timedelta"):
             ser.cumprod()
+
+    @pytest.mark.parametrize(
+        "data, op, skipna, expected_data",
+        [
+            ([], "cumsum", True, []),
+            ([], "cumsum", False, []),
+            (["x", "z", "y"], "cumsum", True, ["x", "xz", "xzy"]),
+            (["x", "z", "y"], "cumsum", False, ["x", "xz", "xzy"]),
+            (["x", pd.NA, "y"], "cumsum", True, ["x", "x", "xy"]),
+            (["x", pd.NA, "y"], "cumsum", False, ["x", pd.NA, pd.NA]),
+            ([pd.NA, "x", "y"], "cumsum", True, ["", "x", "xy"]),
```

Comment:
It seems that for numerical data, we actually (somewhat inconsistently?) propagate _leading_ NAs:

```
In [7]: pd.Series([np.nan, 0.5, 2.5]).cumsum()
Out[7]: 
0    NaN
1    0.5
2    3.0
dtype: float64
```

(i.e. the result doesn't have 0.0 for the first element)

Reply:
Actually not related to "leading" NAs. It seems what is happening is that missing values are ignored to calculate the cumulative result, but then are propagated to the result elementwise. This is also shown in the docstring example of [cumsum](https://pandas.pydata.org/docs/reference/api/pandas.Series.cumsum.html), so this seems intentional.

Reply:
Thanks for catching this. Agreed we should match this behavior. I do find it odd, but that's (possibly) for another day!